Files, scripts and documentation for the LDCS system

floridop opened the pull request:
This PR contains the first attempt to streamline the LDCS system deployment.

This first set of commits contains the needed to create and manage the rucio server.

It also contains a basic set of tools to create and manage GRID certificates for the infrastructure.

All is contained in the `maintenance/` folder.

floridop committed:
First commit of maintenance tools

floridop committed:
Update README.md

Cosmetics

floridop committed:
Cosmetics on readme

floridop committed:
Added docker compose file for rucio server

floridop committed:
Added default configuration files for rucio-server

floridop committed:
attempt to start rucio daemons in the same docker compose file

floridop committed:
updated README with info about daemons and stopping all services

floridop committed:
Fixed rucio-daemons container version and changed log path to host instead of inside docker

- latest rucio-daemons container seem to have some python dependency problem, must be tested if fixing version helps
- decided to write all logs to host system on /var/log/rucio, to minimize the use of docker volumes.

tvami commented on maintenance/configFiles/rucio-server/alembic.ini:
```diff
@@ -0,0 +1,68 @@
+
+[alembic]
+# path to migration scripts
+script_location = /usr/lib/python2.7/site-packages/rucio/db/sqla/migrate_repo/
```
is there no py3 version? I'm worried for the fact that py2 is deprecated...

floridop replied:
Fair point, I am investigating if it is possible to use newer versions. Right now we are in the process of resuming an old setup, so maybe this will be improved. For now I believe the containers and systems that were used still had python2.7, and maybe this migration was already done, probably that line is not needed anymore. I'll report when I progress.

floridop replied:
The latest commit has all systems updated to python3.6.

floridop committed:
Upgraded to rucio version 1.26.11

- Removed WSGI configuration, seems not to be needed anymore.
- Rewieved log output for both server and daemons
- Changed alembic.ini to support a certain version of python3 in the containers

Known issues:
  - abacus logs errors about not finding the database. I suspect
it requires a different configuration.
  - sometimes daemons don't start or die abruptly.

bryngemark commented:
@floridop happy with this for now?

floridop committed:
Definine compose dependencies and restart policies

floridop commented:
Hi all
I see the changes where approved but never merged, now I am working on this again, so I will keep working on the same branch.

floridop committed:
Helper files to create/ship the certficate bundle using tar
